images/devcontainer/features.md

```diff
@@ -53,6 +53,9 @@ may depend on which method you wish to use to install the R binary packages.
 
 Some tips for choosing:
 
+- When installing R packages via `apt` or `mamba` (`conda`), dependencies are installed automatically.
+  But, installing R packages via R function (`install.packages`)
+  may require separate `apt` installations of system libraries that are dependencies.
```

Comment:
Not so!  And that is another key whole point of `r2u`:

```sh
edd@rob:~$ docker run --rm -ti eddelbuettel/r2u:jammy Rscript -e 'install.packages("RPostgreSQL")'        
Install system packages as root...                                                                       
Reading package lists... Done                                                                            
Building dependency tree... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Hit http://archive.ubuntu.com/ubuntu jammy InRelease                                                                                                                                                              
Ign https://r2u.stat.illinois.edu/ubuntu jammy InRelease                                                                                                                                                          
Get:1 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [114 kB]                                                                                                                                           
Get:2 https://r2u.stat.illinois.edu/ubuntu jammy Release [5713 B]                                                                                                                                                 
Get:3 https://r2u.stat.illinois.edu/ubuntu jammy Release.gpg [793 B]                                                                                                                                              
Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]                                                                                                                                         
Get:5 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [99.8 kB]                                                                                                                                        
Get:6 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2179 kB]                                                                                                                                    
Get:7 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [6760 kB]                                                                                                                                      
Get:8 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [946 kB]                                                                                                                             
Get:9 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease [18.1 kB]                                                                                                                      
Get:10 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [568 kB]                                                                                                                          
Get:11 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [903 kB]                                                                                                                                
Get:12 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [8097 B]                                                                                                                          
Get:13 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [7277 B]                                                                                                                          
Get:14 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [764 kB]                                                                                                                          
Get:15 https://ppa.launchpadcontent.net/edd/misc/ubuntu jammy InRelease [18.1 kB]                                                                                                                                 
Get:16 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [584 kB]                                                                                                                              
Get:17 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [486 kB]                                                                                                                        
Get:18 https://ppa.launchpadcontent.net/marutter/rrutter4.0/ubuntu jammy InRelease [17.5 kB]                                                                                                                      
Get:19 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 Packages [974 kB]                                                                                                            
Get:20 https://ppa.launchpadcontent.net/edd/misc/ubuntu jammy/main amd64 Packages [5250 B]                                                                                                                        
Get:21 https://ppa.launchpadcontent.net/marutter/rrutter4.0/ubuntu jammy/main amd64 Packages [7604 B]                                                                                                             
Fetched 14.6 MB in 0s (0 B/s)                                                                                                                                                                                     
Reading package lists... Done    
Building dependency tree... Done 
Reading state information... Done
Get:1 https://r2u.stat.illinois.edu/ubuntu jammy/main all r-cran-dbi all 1.1.3-1.ca2204.1 [697 kB]                                                                                                                
Get:2 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 r-cran-rpostgresql amd64 0.7-4-1.ca2204.1 [349 kB]                                                                                                    
Get:3 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libpq5 amd64 14.5-0ubuntu0.22.04.1 [140 kB]                                                                                                       
Fetched 1186 kB in 0s (0 B/s)                                                                                                                                                                                     
Selecting previously unselected package libpq5:amd64.
(Reading database ... 21782 files and directories currently installed.)
Preparing to unpack .../libpq5_14.5-0ubuntu0.22.04.1_amd64.deb ...
Unpacking libpq5:amd64 (14.5-0ubuntu0.22.04.1) ...
Selecting previously unselected package r-cran-dbi.
Preparing to unpack .../r-cran-dbi_1.1.3-1.ca2204.1_all.deb ...
Unpacking r-cran-dbi (1.1.3-1.ca2204.1) ...
Selecting previously unselected package r-cran-rpostgresql.
Preparing to unpack .../r-cran-rpostgresql_0.7-4-1.ca2204.1_amd64.deb ...
Unpacking r-cran-rpostgresql (0.7-4-1.ca2204.1) ...
Setting up r-cran-dbi (1.1.3-1.ca2204.1) ...
Setting up libpq5:amd64 (14.5-0ubuntu0.22.04.1) ...
Setting up r-cran-rpostgresql (0.7-4-1.ca2204.1) ...
Processing triggers for libc-bin (2.35-0ubuntu3.1) ...
edd@rob:~$ 
```

Moreover, suppose PostgreSQL changes their API and the library becomes `libpq6`.  _Every other system will now potentially uninstall `libpq5` and break the `RPostgreSQL` we just added._  But with `r2u` and *system-level integration* `apt` now knows about the "client and user" of `libpq5` and will *not* uninstall that.   

If you care about your system stability, this matters.  No separate package installer that is not integrated with the system (`box`, `renv`; `pak`, `rig`, ...) can do this.

Reply:
Your example is a case of configuring bspm, right?
I understand that, but so far I don't think the r-apt Feature works that way because it doesn't configure bspm now.

Reply:
Not really, rather the inverse.  The system integration is `r2u` and an *apt*-callable repo. `bspm` is a very nice cherry on top that gives you `install.packages()` from R instead of the the `sudo apt install r-cran-*` we did all those years with c2d4u.

`bspm` _extends_ `r2u` but the core features ("cheap", "fast", "reliable" access to 20k CRAN packages as binaries integrated into the system) of `r2u` do not depend on or use `bspm`.

Reply:
Sorry, but I don't understand your main point.
What does your "Not so" mean? (I interpreted it as a point to the explanation I wrote about install.packages)

BTW, here is only description as it relates to the Dev Container Features.
`r-apt` means `ghcr.io/rocker-org/devcontainer-features/r-apt`, not `docker.io/rocker/r-apt`.